diff --git a/drivers/gpu/drm/eswin/es_dc_hw.c b/drivers/gpu/drm/eswin/es_dc_hw.c
index 30bcfb6b759c..c453702c2b80 100644
--- a/drivers/gpu/drm/eswin/es_dc_hw.c
+++ b/drivers/gpu/drm/eswin/es_dc_hw.c
@@ -1211,7 +1211,7 @@ static void plane_ex_commit(struct dc_hw *hw)
 				if (i == PRIMARY_PLANE)
 					dc_set_clear(hw,
 						     DC_FRAMEBUFFER_CONFIG_EX,
-						     BIT(6), BIT(8));
+						     0, BIT(8));
 				else
 					dc_set_clear(hw, DC_OVERLAY_CONFIG,
 						     BIT(29), BIT(30));
